UIforETW/UIforETWDlg.cpp

```diff
@@ -40,10 +41,821 @@ const int kRecordTraceHotKey = 1234;
+
+const wchar_t SampledStacksString[ ] = 
+	L"This enables recording of call stacks on CPU sampling events, which "
+	L"by default happen at 1 KHz. This should rarely be disabled.";
+
+const wchar_t SampleRateString[ ] = 
+	L"Checking this changes the CPU sampling frequency from the default of "
+	L"~1 KHz to the maximum speed of ~8 KHz. "
+	L"This increases the data rate and thus the size of traces "
+	L"but can make investigating brief CPU-bound "
+	L"performance problems (such as a single long frame) "
+	L"more practical.";
+
+const wchar_t GPUTracingString[ ] =
+	L"Check this to allow seeing GPU usage "
+	L"in WPA, and more data in GPUView.";
+
+const wchar_t ShowCommandsString[ ] =
+	L"This tells UIforETW to display the commands being "
+	L"executed. This can be helpful for diagnostic purposes "
+	L"but is not normally needed.";
+
+const wchar_t TracingModeString[ ] =
+	L"Select whether to trace straight to disk or to in-memory circular buffers.";
+
+const wchar_t TracesString[ ] =
+	L"This is a list of all traces found in %etwtracedir%, which defaults to "
+	L"documents\\etwtraces.";
+
+const wchar_t TraceNotesString[ ] =
+	L"Trace notes are intended for recording information about ETW traces, such "
+	L"as an analysis of what was discovered in the trace. "
+	L"Trace notes are auto-saved to a parallel text "
+	L"file - just type your analysis. "
+	L"The notes files will be renamed when you rename traces "
+	L"through the trace-list context menu.";
+
+const char NtSymbolEnvironmentVariableName[ ] =
+	"_NT_SYMBOL_PATH";
+
+const char NtSymCacheEnvironmentVariableName[ ] =
+	"_NT_SYMCACHE_PATH";
+
+const char DefaultSymbolPath[ ] =
+	"SRV*c:\\symbols*http://msdl.microsoft.com/download/symbols";
+
+const char ChromiumSymbolPath[ ] =
+	"SRV*c:\\symbols*http://msdl.microsoft.com/download/symbols;"
+	"SRV*c:\\symbols*https://chromium-browser-symsrv.commondatastorage.googleapis.com";
+
+const char DefaultSymbolCachePath[ ] =
+	"c:\\symcache";
+
+void handleUnexpectedCreateDirectory( _In_ const std::wstring& path, _In_ const DWORD lastErr )
+{
+	OutputDebugStringA( "UIforETW: Encountered an unexpected error after "
+						"calling CreateDirectory!\r\n"
+						"UIforETW:\tAttempting to create folder:" );
+	OutputDebugStringW( path.c_str( ) );
+	OutputDebugStringA( "\r\n" );
+	ErrorHandling::outputErrorDebug( lastErr );
+	std::terminate( );
+}
+
+//Compiler didn't eliminate code for std::wstring
+//Seems to generate better code when passed a PCWSTR for path.
+void dieUnexpectedErrorPathFileExists( _In_z_ PCWSTR const path, _In_ const DWORD lastErr )
+{
+	OutputDebugStringA("UIforETW: Encountered an unexpected error after calling PathFileExists!\r\n");
+	OutputDebugStringA("\tPath that we were checking: \r\n\t");
+	OutputDebugStringW(path);
+	OutputDebugStringA("\r\n");
+	ErrorHandling::outputErrorDebug(lastErr);
+	std::terminate();
+}
+
+//Compiler didn't eliminate code for std::wstring
+//Seems to generate better code when passed a PCWSTR for path.

```

Comment:
Why are you micro-optimizing this? const ref string should be the standard. There is no need to optimize beyond what is clearest, safest, and most consistent.